src/app.js

```diff
@@ -279,13 +296,15 @@ export function app(app) {
           removeElement(element, reusableChild[0], reusableNode)
         }
       }
+      recoverNode(oldNode)
```

Comment:
@ngryman @jbucaran I think this catches the oldNodes that need to be collected that I had missed by only collecting in the removeElement(),  I think this is easiest way to not step on reusableNodes for the keyed updates while still recycling.  Really my naive initial recursive recycle in render() that I commented on above just causes an extra iteration of all the nodes instead of hooking in during the patch to recycle like I tried in this PR.  But I did miss this recoverNode(oldNode)  I may have missed another, or there maybe a better place to do all of this, if this PR is closed/rejected I won't mind it's serving it's purpose by getting comments/review as far as I'm concerned :slightly_smiling_face: